client/state/initial-state.js

```diff
@@ -28,25 +29,25 @@ const HOUR_IN_MS = 3600000;
 export const SERIALIZE_THROTTLE = 5000;
 export const MAX_AGE = 7 * DAY_IN_HOURS * HOUR_IN_MS;
 
-function getInitialServerState() {
-	// Bootstrapped state from a server-render
-	if ( typeof window === 'object' && window.initialReduxState && ! isSupportUserSession() ) {
-		const serverState = reducer( window.initialReduxState, { type: DESERIALIZE } );
-		return pick( serverState, Object.keys( window.initialReduxState ) );
-	}
-	return {};
-}
-
-function serialize( state ) {
+function serialize( state, reducer ) {
 	const serializedState = reducer( state, { type: SERIALIZE } );
 	return Object.assign( serializedState, { _timestamp: Date.now() } );
 }
 
-function deserialize( state ) {
+function deserialize( state, reducer ) {
```

Comment:
that's.... going to be interesting.

Reply:
Yes! Deserializing a state fragment for, e.g., `reader` must run the persisted object through the `reader` reducer, not the global one. It all works together very nicely in #27415.